### PR TITLE
Entitled Badges - Role Binding

### PR DESCRIPTION
**Pull request by AnatoleAM**
Adding a `RoleBinding` property to badge entitlement data. This allows a badge entitlement to be directly bound to a role or role entitlement, essentially an extra permission check. The current use case for this in production is to hide sub badges for users who are no longer subscribers

### CODE REVIEW

File: src/server/api/v2/gql/resolvers/mutation/entitlement.go [hunk 1]
```diff
@@ -50,6 +50,11 @@ func (*MutationResolver) DeleteEntitlement(ctx context.Context, args struct {
 	}, nil
 }
 
+func (*MutationResolver) EditEntitlement(ctx context.Context, args struct {
+}) (*response, error) {
+	return nil, nil
```
Comment (TroyDota) [inline src/server/api/v2/gql/resolvers/mutation/entitlement.go:55]: why is this unimplemented?

**Review by TroyDota — CHANGES_REQUESTED** [review]
Where is the change to the GQL schema.

File: src/server/api/v2/gql/resolvers/mutation/entitlement.go (hunk 1 above)
Comment (AnatoleAM) [reply src/server/api/v2/gql/resolvers/mutation/entitlement.go:55]: Forgot to remove this

**Comment by AnatoleAM** [comment]
Task linked: [CU-1g5h07v Badge Entitlement: Role Binding](https://app.clickup.com/t/1g5h07v)

**Pull request merged**